examples/nemo/scripts/kernels_trans.py

```diff
@@ -0,0 +1,732 @@
+# -----------------------------------------------------------------------------
```

Comment:
Wow, that's a big script file, although you did warn me :-). As agreed I am not going to suggest changes to this as the point of this PR is to get it onto master so it is easier to maintain. As expected there are lots of good things in it which we should be able to generalise and provide as useful functionality to users over time.